lms/djangoapps/ccx/models.py

```diff
@@ -16,6 +26,71 @@ class CustomCourseForEdX(models.Model):
     display_name = models.CharField(max_length=255)
     coach = models.ForeignKey(User, db_index=True)
 
+    _course = None
+    _start = None
+    _due = _MARKER
+
+    @property
+    def course(self):
+        if self._course is None:
+            store = modulestore()
+            with store.bulk_operations(self.course_id):
+                course = store.get_course(self.course_id)
+                if course and not isinstance(course, ErrorDescriptor):
+                    self._course = course
+                else:
+                    log.error("CCX {0} from {2} course {1}".format(  # pylint: disable=logging-format-interpolation
+                        self.display_name, self.course_id, "broken" if course else "non-existent"
+                    ))
+
+        return self._course
+
+    @property
+    def start(self):
+        if self._start is None:
```

Comment:
BTW, we use a third-party package called lazy to create compute-once properties like this without the boilerplate:
```
@lazy
def start(self):
    from .overrides import get_override_for_ccx
    return get_override_for_ccx(self, self.course, 'start')
```